pkg/tfbridge/provider.go

```diff
@@ -780,18 +780,24 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	rescfg := MakeTerraformConfigFromInputs(ctx, p.tf, inputs)
 	warns, errs := p.tf.ValidateResource(ctx, tfname, rescfg)
 	for _, warn := range warns {
-		if err = p.host.Log(ctx, diag.Warning, urn, fmt.Sprintf("%v verification warning: %v", urn, warn)); err != nil {
-			return nil, err
+		warning := fmt.Sprintf("%v verification warning: %v", urn, warn)
+		if p.host == nil {
```

Comment:
Probably a code-level TODO here that this temporarily required for tests, before we can inject a mock host in the tests.. Ideally we'd not make the prod code more complex for tests.